src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1553,8 +1569,15 @@ func (k *kataAgent) handleDeviceBlockVolume(c *Container, m Mount, device api.De
 		vol.Driver = kataBlkCCWDevType
 		vol.Source = blockDrive.DevNo
 	case c.sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlock:
-		vol.Driver = kataBlkDevType
-		vol.Source = blockDrive.PCIPath.String()
+		driveroption := k.CheckandSetDmVerityDevice(m)
```

Comment:
We should handle all block device here instead only `VirtioBlock`.
Logically we add an `dm-verity` device over a block device. So let's add the logic to handle dm-verity type after line 1589, to convert a block device into a dm-verity device when needed.

Reply:
Done.